swift/StableDiffusion/pipeline/Encoder.swift

```diff
@@ -0,0 +1,79 @@
+// For licensing see accompanying LICENSE.md file.
+// Copyright (C) 2022 Apple Inc. All Rights Reserved.
+
+import Foundation
+import CoreML
+
+@available(iOS 16.0, macOS 13.0, *)
+/// Encoder, currently supports image2image
+public struct Encoder {
+    
+    public enum Error: String, Swift.Error {
+        case latentOutputNotValid
+        case batchLatentOutputEmpty
+    }
+    
+    /// VAE encoder model + post math and adding noise from schedular
+    var model: MLModel
```

Comment:
Could you please adopt the [ManagedMLModel](https://github.com/apple/ml-stable-diffusion/blob/main/swift/StableDiffusion/pipeline/ManagedMLModel.swift#L11) class so this model can also be offloaded when required?